### ISSUE TITLE
Prune failed - chunk not found

### ISSUE THREAD

**Issue by kairisku** [gilbertchen/duplicacy#386]
Prune just failed with the following error:

```
# tail -5 20180320_prune.txt
Deleting snapshot Caveman at revision 28
Deleting snapshot Caveman at revision 29
Deleting snapshot Hotblack at revision 28
Deleting snapshot Hotblack at revision 29
Chunk cf9557b4daf46117a87b1e844579a3a848e8771c3ae829efc3d7ddc0fee1a018 can't be found
```

As per the debugging instructions, I searched the prune logs for any mentions of the missing chunk, and the only reference was from the exact same prune run where it was just deleted:

```
# tail -5 prune-log-20180320-071539
Deleted chunk 75b704808d835fd5ebe33830378546837e5761919aed38887163784aded0a825 (exclusive mode)
Deleted chunk cf9557b4daf46117a87b1e844579a3a848e8771c3ae829efc3d7ddc0fee1a018 (exclusive mode)
Deleted chunk e949396a8b7b4a1be8c381aed68dd4aff4ec02615c5f912c65611874d750301f (exclusive mode)
Deleted chunk 6f8ad19227e2b7751484ceca0be47ae613eb5beb6601fe0a198a29941afb6da5 (exclusive mode)
Deleted chunk a89d6aaa589ce84e7cfd8ba7f60f0161e6f7668dcde4ac8ef60a27002753f050 (exclusive mode)
```

Checking the timestamps of the duplicacy -produced prune log versus the stdout redirected to a file gives about the same result:

```
# stat prune-log-20180320-071539
  File: `prune-log-20180320-071539'
  Size: 124416          Blocks: 248        IO Block: 32768  regular file
Device: 17h/23d Inode: 283772182   Links: 1
Access: (0600/-rw-------)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2018-03-20 08:43:25.713779940 +0200
Modify: 2018-03-20 07:16:43.792074064 +0200
Change: 2018-03-20 07:16:43.000000000 +0200
# stat 20180320_prune.txt
  File: `20180320_prune.txt'
  Size: 723             Blocks: 8          IO Block: 32768  regular file
Device: 17h/23d Inode: 324535188   Links: 1
Access: (0644/-rw-r--r--)  Uid: (    0/    root)   Gid: (    0/    root)
Access: 2018-03-20 08:42:26.255523271 +0200
Modify: 2018-03-20 07:16:43.777074252 +0200
Change: 2018-03-20 07:16:43.000000000 +0200
```

So to me it looks like the prune operation first successfully deleted the chunk and then a few milliseconds later tried to delete it again and failed. No, there are no overlapping runs, because I have exactly one cron job that first runs a few backups and then does a prune. So what is going on here?

**Comment by gilbertchen** [comment]
This seems to have been fixed by #329, which I just merged into the master branch.

What happened here was that there was a chunk that contained the list of chunk hashes shared by two snapshots.  If these two snapshots were to be deleted at the same time, this chunk would be deleted by the first snapshot but then the second snapshot still needed this chunk in order to get the list of chunk hashes.
